webpack/assets/javascripts/react_app/components/common/table/components/TemplateActionButton.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../../common/I18n';
+import { ActionButtons } from '../../ActionButtons/ActionButtons';
+
+export const TemplateActionButton = ({
+  id,
+  name,
+  vendor,
+  availableActions: { generatable, lockable, unlockable },
+  templateActions,
+  permissions: { canDelete },
+}) => {
+  const buttons = [];
+  if (generatable) {
+    buttons.push({
+      title: __('Generate'),
+      action: {
```

Comment:
This file is a bit messy to read
you can use spread (`...`) for building a conditional array, much shorter :)
i.e
```js
const teplateButtons = [
  ...(generatable ? [{title: 'Generate', action: {}}] : []),
  {title: "Clone", action: {}},
  {title: "Export", action: {}},
  ...(lockable ? [{title: 'Lock', action: {}}] : [])
];
```